packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -47,6 +47,12 @@ typedef FrameCallback = void Function(Duration timeStamp);
 /// task does not return a value.
 typedef TaskCallback<T> = T Function();
 
+/// Signature for [SchedulerBinding.scheduleAsyncTask] callbacks.
+///
+/// The type argument `T` is the task's return value. Consider `void` if the
+/// task does not return a value.
+typedef AsyncTaskCallback<T> = Future<T> Function();
```

Comment:
FYI, there is already `AsyncValueGetter` over in foundation with the exact same signature: https://api.flutter.dev/flutter/foundation/AsyncValueGetter.html